kfs-core/src/main/java/org/kuali/kfs/vnd/document/validation/impl/VendorRule.java

```diff
@@ -233,12 +238,48 @@ private boolean processValidation(MaintenanceDocument document) {
             valid &= processAddressValidation(document);
             valid &= processContractValidation(document);
             valid &= processCommodityCodeValidation(document);
+            valid &= processSupplierDiversityValidation(document);
         }
 
         return valid;
     }
 
-    /**
+    private boolean processSupplierDiversityValidation(MaintenanceDocument document) {
+    	boolean valid = true;
+    	List<VendorSupplierDiversity> vendorSupplierDiversities = ((VendorDetail) document.getNewMaintainableObject().getBusinessObject()).getVendorHeader().getVendorSupplierDiversities();
+    	DataDictionaryService dataDictionaryService = SpringContext.getBean(DataDictionaryService.class);
+    	
+    	if(vendorSupplierDiversities == null || vendorSupplierDiversities.isEmpty()) {
+    		valid = false;
+    		String fieldName = VENDOR_SUPPLIER_DIVERSITY_CODE;
+    		String attributeLabel = dataDictionaryService.getAttributeLabel(VendorSupplierDiversity.class, fieldName);
+    		String shortLabel = dataDictionaryService.getAttributeShortLabel(VendorSupplierDiversity.class, fieldName);
+    		
+    		putFieldError(VendorPropertyConstants.VENDOR_SUPPLIER_DIVERSITY_CODE, RiceKeyConstants.ERROR_REQUIRED, attributeLabel + " (" + shortLabel + ")");
+    	}
+    	
+    	else {
```

Comment:
This validation for active diversity information was not included in the functional spec. It also should not be needed. The available list of supplier diversities should only include active diversity values.